packages/splits-sdk/src/client/data.ts

```diff
@@ -0,0 +1,1061 @@
+      }
+    }
+
+    const internalBalances = formatAccountBalances(response.balances)
+    if (response.type === 'user') {
+      // Only including split main balance for users
+      return {
+        withdrawn,
+        distributed,
+        activeBalances: internalBalances,
+      }
+    }
+
+    // Need to fetch current balance. Handle alchemy/infura with logs, and all other rpc's with token list
+    if (!this._publicClient)
+      throw new MissingPublicClientError(
+        'Public client required to get active balances. Please update your call to the client constructor with a valid public client, or set includeActiveBalances to false',
+      )
+    const tokenList = erc20TokenList ?? []
+
+    let balances: TokenBalances
+    if (
+      erc20TokenList === undefined &&
+      isAlchemyPublicClient(this._publicClient)
+    ) {
+      // If no token list passed in and we're using alchemy, fetch all balances with alchemy's custom api
+      balances = await fetchContractBalancesWithAlchemy(
+        accountAddress,
+        this._publicClient,
+      )
+    } else {
+      if (erc20TokenList === undefined) {
+        // If no token list passed in, make sure the public client supports logs and then fetch all erc20 tokens
+        if (!isLogsPublicClient(this._publicClient))
+          throw new InvalidArgumentError(
+            'Token list required if public client is not alchemy or infura',
+          )
+        const transferredErc20Tokens = await fetchERC20TransferredTokens(
+          chainId,
+          this._publicClient,
+          accountAddress,
+        )
+        tokenList.push(...transferredErc20Tokens)
+      }
+
+      // Include already distributed tokens in list for balances
+      const customTokens = Object.keys(distributed) ?? []
+      const fullTokenList = Array.from(
+        new Set(
+          [zeroAddress, ...tokenList]
+            .concat(Object.keys(internalBalances))
+            .concat(customTokens)
+            .map((token) => getAddress(token)),
+        ),
+      )
+      balances = await fetchActiveBalances(
+        accountAddress,
+        this._publicClient,
+        fullTokenList,
+      )
+    }
+
+    const allTokens = Array.from(
+      new Set(Object.keys(balances).concat(Object.keys(internalBalances))),
+    )
+    const filteredBalances = allTokens.reduce((acc, token) => {
+      const internalBalanceAmount = internalBalances[token] ?? BigInt(0)
+      const contractBalanceAmount = balances[token] ?? BigInt(0)
+
+      // SplitMain leaves a balance of 1 for gas efficiency in internal balances.
+      // Splits leave a balance of 1 (for erc20) for gas efficiency
+      const tokenBalance =
+        (internalBalanceAmount > BigInt(1)
+          ? internalBalanceAmount
+          : BigInt(0)) +
+        (contractBalanceAmount > BigInt(1) ? contractBalanceAmount : BigInt(0))
+      if (tokenBalance > BigInt(0)) acc[token] = tokenBalance
+
+      return acc
+    }, {} as TokenBalances)
```

Comment:
this is not including the warehouse balance, right? do we want to include that?

Reply:
yes for now just kept stuff as is, didn't think we need to add it rn so left it out